File: be_fastapi/app/schemas/violation.py
from pydantic import BaseModel
from typing import Optional, Any
from uuid import UUID


class ViolationResponse(BaseModel):
    id: UUID
    source_id: UUID
    rule_id: UUID
    timestamp: str
    detected_license_plate: Optional[str] = None
    evidence_url: str
    metadata: Any = {}

    class Config:
        from_attributes = True


class ViolationListResponse(BaseModel):
    violations: list[ViolationResponse]
    total: int
    skip: int
    limit: int

File: fe_react/src/types/violation.ts
export interface Violation {
  id: string;
  source_id: string;
  rule_id: string;
  timestamp: string;
  detected_license_plate: string | null;
  evidence_url: string; // URL to image/video
  metadata: Record<string, any>;
}

export interface ViolationListResponse {
  violations: Violation[];
  total: number;
  skip: number;
  limit: number;
}
